### PR TITLE
Add support for indexed CSCs

### PR DESCRIPTION
**Pull request by SimonKrughoff**
I'm not sure this is the entire answer, but I think it may be close.  @mareuter will you please have a look?

### CODE REVIEW

File: lsst_efd_client/efd_helper.py [hunk 1]
```diff
@@ -98,8 +98,10 @@ def build_time_range_query(self, topic_name, fields, start, end, is_window=False
         else:
             raise TypeError('The second time argument must be the time stamp for the end ' +
                             'or a time delta.')
-
-        timespan = f"time >= '{start_str}Z' AND time <= '{end_str}Z'"  # influxdb demands the trailing Z
+        index_str = ''
+        if index:
+            index_str = f' AND {topic_name}ID = {index}'
```
Comment (mareuter) [inline lsst_efd_client/efd_helper.py]: This isn't going to work. You need to parse out the CSC name. Looking at your example notebooks, if you use this topic, _lsst.sal.ATCamera.wreb_, you'll get _lsst.sal.ATCamera.wrebID_ which isn't the expected _ATCameraID_ field.

Comment (SimonKrughoff) [reply lsst_efd_client/efd_helper.py]: You are absolutely right.  I was too hasty creating this pull request.  Sorry about that.  I'll try to figure out the best way to parse out the CSC name.

Comment (SimonKrughoff) [reply lsst_efd_client/efd_helper.py]: OK, I've now tested this with one indexed CSC and it seemed to do the right thing.  I also include another commit which prevents saving creds in the constructed object.  This isn't too risky, but if the object were pickled, someone could possibly retrieve them.

Comment (mareuter) [reply lsst_efd_client/efd_helper.py]: Good catch on the credential caching.

**Pull request merged**